rust/src/dns/dns.rs

```diff
@@ -732,14 +748,28 @@ fn probe_header_validity(header: &DNSHeader, rlen: usize) -> (bool, bool, bool)
         return (false, false, false);
     }
 
+    if (header.additional_rr as usize
+        + header.answer_rr as usize
+        + header.authority_rr as usize
+        + header.questions as usize) == 0 && rlen > DNS_HEADER_SIZE {
+            // zero fields, data size should be just DNS_HEADER_SIZE
+            // happens when DNS server return s format error
+            return (false, false, false);
+        }
     let is_request = header.flags & 0x8000 == 0;
+    if dir == Direction::ToClient && is_request && rlen > 0x1000 {
```

Comment:
Also, shoud we check the direction here? The direction could be flipped based on the return value of this function.

Reply:
That is a good question :
My point is to gather 2 hints that this is not DNS : direction is flipped, and record length is big (as the comment states below)

Do you have real cases in DNS where the direction is actually flipped ?

Reply:
And I guess my question to you is how would you handle the private pcaps ?

Reply:
> Do you have real cases in DNS where the direction is actually flipped ?

This can happen when there is packet loss and the request is lost, the response will be picked up as a request I believe. The S-V test `dns-reversed-udp-1` shows this "flow" flipping in action - however, lacks the test to test it, which should probably be added.

Reply:
PR here: https://github.com/OISF/suricata-verify/pull/2062/files